js/components/chart-base.tsx

```diff
@@ -165,7 +165,9 @@ class ChartBase extends React.Component<IChartBaseProps, null> {
     }
 
     componentWillReceiveProps(nextProps) {
```

Comment:
componentWillReceiveProps is an old react lifecycle method.
Could you replace this with componentDidUpdate and test.
More info: https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops